### PR TITLE
Provide initial setup to restore snapshots from BE tests

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

This adds a very minimal set of functions to load a snapshot from the Backend Integration test suite.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204274556967336/1204549651666220/f) _(internal)_

## 🔍 What does this change?

- Add a `subgraph` directory in the test suite, were I plan to insert new subgraph tests
- Provide a function to easily restore a snapshot (using the `status` library for errors)
- Add a simple test case to insert an empty snapshot, which only contains the metadata.

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

 - [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] affected the execution graph as it adds a dependency for the BE test suite on `hash-graph`, however, `turbo.json` does not need to be adjusted.

### CODE REVIEW

File: tests/hash-backend-integration/src/tests/setup.ts [hunk 1]
```diff
@@ -16,3 +19,24 @@ export const recreateDbAndRunSchemaMigrations = async () => {
...
+      fetch("http://localhost:4001/snapshot", {
+        method: "POST",
+        body: snapshot,
+      }),
+    )
+    .then(async (response) => {
+      const status: GraphStatus = await response.json();
+      if (status.code !== "OK") {
+        if (status.message) {
+          throw new Error(`Snapshot restoration error: ${status.message}`);
```
Comment (Alfred-Mountfield) [inline tests/hash-backend-integration/src/tests/setup.ts:36]: Simply printing the message will probably lose a lot of information, such as accompanying `ErrorInfo` structs.

File: tests/hash-backend-integration/src/tests/subgraph/simple.jsonl
Comment (Alfred-Mountfield) [inline tests/hash-backend-integration/src/tests/subgraph/simple.jsonl:1]: Do we perhaps want to call this `empty.jsonl`?

Comment (TimDiekmann) [reply tests/hash-backend-integration/src/tests/subgraph/simple.jsonl:1]: Actually, I wanted to add `empty.jsonl`, which is literally empty and ensure that this is failing but that could be named `empty-misformed.jsonl` or something. Ideas?

Comment (Alfred-Mountfield) [reply tests/hash-backend-integration/src/tests/subgraph/simple.jsonl:1]: Perhaps `valid/empty-state.jsonl`, and the other could be `invalid/empty.jsonl` and we could have other variants of `invalid` kinds

File: tests/hash-backend-integration/src/tests/setup.ts (hunk 1 above)
Comment (Alfred-Mountfield) [reply tests/hash-backend-integration/src/tests/setup.ts:36]: We do have `GraphApiError` you could reuse from the backend, but I agree about doing that in follow-up. I just think we should opt to `JSON.stringify` in the interim

Comment (TimDiekmann) [reply tests/hash-backend-integration/src/tests/setup.ts:44]: I added a logging output: [Log full status object on failure](https://github.com/hashintel/hash/pull/2530/commits/9f4bdeb77fec167f2beda7d72fbc454abe045068)

Comment (Alfred-Mountfield) [reply tests/hash-backend-integration/src/tests/setup.ts:44]: Can we also just put it in the error message for now please. We've had a few reports recently of devs not being able to find the information associated with an error and we don't really want them having to dig through a history of the logs. Especially as this is a test and sometimes messages get swallowed.

**Review by Alfred-Mountfield — APPROVED** [review]

File: tests/hash-backend-integration/src/tests/subgraph/simple.jsonl
Comment (TimDiekmann) [reply tests/hash-backend-integration/src/tests/subgraph/simple.jsonl:1]: I went with `pass` and `fail` as these are common terms for testing.

**Review by Alfred-Mountfield — DISMISSED** [review]
LGTM 🪨 🚀

**Review by Alfred-Mountfield — APPROVED** [review]
